domains/DataCollection/Forms/SupplementaryForm/index.js

```diff
@@ -85,12 +86,7 @@ const SupplementaryForm = ({
           };
         }
 
-        postObjectsToClassWithRelation(postParams).then(() => {
-          setTimeout(() => {
-            actions.setSubmitting(false);
-            toRoot();
-          }, 1000);
-        });
+        postSupplementaryForm(postParams, actions, toRoot);
```

Comment:
read my comment below, but it makes more sense to keep all Formik logic in this page

Reply:
think this should be taken care of now.